### PR TITLE
Change branches to branches/tags 

### LINKED ISSUE #556: using cache on tags does not work
hi! 

I have a cache created on a non-default branch, all works fine except when the workflow is run on **a tag**, the cache isn't hit 

when it's not a tagged commit, and my github.ref is "...heads" it starts working again.

I guess it's related to https://github.com/actions/cache#cache-scopes

Essentially, it looks like there is no way to use cache in runs on tagged revisions, or is it? (I don't have any caches on my master yet). Is it an intended behavior? 

At least, it'd be good if it was described in readme in that "scopes" section.

### PR DESCRIPTION
**Pull request by Phantsure**
### Why:

Closes https://github.com/actions/cache/issues/556
cc @bishal-pdMSFT 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Cache cannot be shared between sibling branches and tags. The documentation doesn't account for `tags`.
Note: There is a Japanese translation as well for this document. A little unsure how that can be updated 

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ For more information on workflow run artifacts, see "[Persisting workflow data u
 
 A workflow can access and restore a cache created in the current branch, the base branch (including base branches of forked repositories), or the default branch (usually `main`). For example, a cache created on the default branch would be accessible from any pull request. Also, if the branch `feature-b` has the base branch `feature-a`, a workflow triggered on `feature-b` would have access to caches created in the default branch (`main`), `feature-a`, and `feature-b`.
 
-Access restrictions provide cache isolation and security by creating a logical boundary between different branches. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`).
+Access restrictions provide cache isolation and security by creating a logical boundary between different branches/tags. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`).
```
Comment (bishal-pdMSFT) [inline content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md:54]:
```suggestion
Access restrictions provide cache isolation and security by creating a logical boundary between different branches/tags. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`). On similar lines, if a cache created for tag `release-a` (from the base `main`) would not be accessible to a workflow triggered for tag `release-b` (with the base `main`)
```

**Comment by Sakhilemasina** [comment]
Access

File: content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md (hunk 1 above)
Comment (Phantsure) [reply content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md:54]: Sounds good

**Comment by cmwilson21** [comment]
👋 @Phantsure Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by PIYUSH-MISHRA-00 — COMMENTED** [review]
included tags

File: content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ For more information on workflow run artifacts, see "[Persisting workflow data u
...
 
-Access restrictions provide cache isolation and security by creating a logical boundary between different branches. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`).
+Access restrictions provide cache isolation and security by creating a logical boundary between different branches/tags. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`). On similar lines, a cache created for tag `release-a` (from the base `main`) would not be accessible to a workflow triggered for tag `release-b` (with the base `main`)
```
Comment (hubwriter) [inline content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md:54]:
```suggestion
Access restrictions provide cache isolation and security by creating a logical boundary between different branches or tags. For example, a cache created for the branch `feature-a` (with the base `main`) would not be accessible to a pull request for the branch `feature-c` (with the base `main`). On similar lines, a cache created for the tag `release-a` (from the base `main`) would not be accessible to a workflow triggered for the tag `release-b` (with the base `main`).
```

**Review by hubwriter — APPROVED** [review]
@Phantsure - Many thanks for raising this PR.

I'll get this merged/published.

**Pull request merged**